text/0060-tasks-api.md

```diff
@@ -568,6 +568,151 @@ New task types are also added for these operations. `indexCreation`, `indexUpdat
 }
 ```
 
+#### 9. Filtering task resources
+
+The `/tasks` and `index/{indexUid}/tasks` endpoints are filterable by type and status query parameters.
+
+##### 9.1. Query parameters definition
+
+| parameter | type | required | description                         |
+|------|------|----------|----------------------------|
+| status | string | No | Possible values are all the status of a task. By default, when `status` query parameter is not set, all task statuses are returned. |
+| type  | string | No | Possible values are all the types of a task. By default, when `type` is not set in, all task types are returned.  |
+
+##### 9.2. Usages examples
+
+This part demonstrates filtering on `/tasks`, it should be equivalent for `indexes/:uid/tasks`.
+
+---
+
+**No filtering**
+
+`GET` - `/tasks`
+
+```json
+{
+    "results": [
+        {
+            "uid": 1350,
+            "indexUid": "movies",
+            "status": "failed",
+            "type": "documentAddition",
+            ...,
+        },
+        ...,
+        {
+            "uid": 1330,
+            "indexUid": "movies_reviews",
+            "status": "succeeded",
+            "type": "documentDeletion",
+            ...
+        }
+    ],
+    ...
+}
+```
+
+**Filter `tasks` that have a `failed` `status`**
+
+`GET` - `/tasks?status=failed`
+
+```json
+{
+    "results": [
+        {
+            "uid": 1350,
+            "indexUid": "movies",
+            "status": "failed",
+            "type": "documentAddition",
+            ...,
+        },
+        ...,
+        {
+            "uid": 1279,
+            "indexUid": "movies",
+            "status": "failed",
+            "type": "settingsUpdate",
+            ...,
+        }
+    ],
+    ...
+}
+```
+
+**Filter `tasks` that are of `documentAddition` type**
+
+`GET` - `/tasks?type=documentAddition`
+
+```json
+{
+    "results": [
+        {
+            "uid": 1350,
+            "indexUid": "movies",
+            "status": "failed",
+            "type": "documentAddition",
+            ...,
+        },
+        ...,
+        {
+            "uid": 1343,
+            "indexUid": "movies",
+            "type": "succeeded",
+            "type": "documentAddition",
+            ...,
+        }
+    ],
+    ...
+}
+```
+
+- 💡 `status` and `type` can be used together. The two parameters are cumulated and a `AND` operation is performed between the two filters.
```

Comment:
How would we express a `OR` in the future if we wanted to?

Reply:
like I want tasks that are either `enqueued` or `processing`

Reply:
I don't think there is a particular need to go that far but it would be possible by separating the values with commas for a `+`.

e.g. `status=enqueued+processing` or `status=enqueued,processing`.